woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1304,3 +1302,78 @@ def test_select_repetitive(sample_df):
     df_repeat_ltypes = schema_df.ww.select(['PhoneNumber', PhoneNumber, 'phone_number'])
     assert len(df_repeat_ltypes.columns) == 1
     assert set(df_repeat_ltypes.columns) == {'phone_number'}
+
+
+def test_accessor_set_index(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    sample_df.ww.init()
+
+    sample_df.ww.set_index('id')
+    assert sample_df.ww.index == 'id'
+    assert (sample_df.index == [0, 1, 2, 3]).all()
```

Comment:
Maybe update this to use the pattern below where we check that the index is equal to sample_df['id'] values instead of hard-coding the numbers, which would break if the id column in sample df ever got changed for some reason.

Reply:
makes sense to me!